src/measure_theory/measure/measure_space.lean

```diff
@@ -1816,13 +1816,66 @@ begin
   exact eq_univ_of_subset (Union_subset_Union $ λ n, subset_to_measurable μ (s n)) hs
 end
 
+/-- Given two σ-finite measures `μ` and `ν`, `finite_spanning_setsin.add μ ν` is a sequence of
+pairwise disjoint, finite spanning sets in `{s | measurable_set s}` with respect to `μ + ν`. -/
+protected def finite_spanning_sets_in.add
+  (S : μ.finite_spanning_sets_in {s | measurable_set s})
+  (T : ν.finite_spanning_sets_in {s | measurable_set s}) :
+  (μ + ν).finite_spanning_sets_in {s | measurable_set s} :=
+{ set := λ n : ℕ, S.set n.unpair.1 ∩ T.set n.unpair.2,
+  set_mem := λ n, (S.set_mem n.unpair.1).inter (T.set_mem n.unpair.2),
+  finite :=
+    begin
+      intro n,
+      rw [add_apply, ennreal.add_lt_top],
+      exact ⟨lt_of_le_of_lt (measure_mono (set.inter_subset_left _ _)) (S.finite n.unpair.1),
+             lt_of_le_of_lt (measure_mono (set.inter_subset_right _ _)) (T.finite n.unpair.2)⟩,
+    end,
+  spanning := by simp_rw [set.Union_unpair (λ i j, S.set i ∩ T.set j), ← set.inter_Union,
+                          ← set.Union_inter, S.spanning, T.spanning, set.inter_univ] }
+
+/-- Given measures `μ`, `ν` where `ν ≤ μ`, `finite_spanning_sets_in.of_le` provides the induced
+`finite_spanning_set` with respect to `ν` from a `finite_spanning_set` with respect to `μ`. -/
+def finite_spanning_sets_in.of_le (h : ν ≤ μ) {C : set (set α)}
+  (S : μ.finite_spanning_sets_in C) : ν.finite_spanning_sets_in C :=
+{ set := S.set,
+  set_mem := S.set_mem,
+  finite := λ n, lt_of_le_of_lt (le_iff'.1 h _) (S.finite n),
+  spanning := S.spanning }
+
 lemma sigma_finite_of_le (μ : measure α) [hs : sigma_finite μ]
   (h : ν ≤ μ) : sigma_finite ν :=
-begin
-  cases hs.out with C,
-  exact ⟨nonempty.intro ⟨C.set, C.set_mem, λ i,
-    (lt_of_le_of_lt (le_iff'.1 h _) (C.finite i)), C.spanning⟩⟩,
-end
+⟨hs.out.map $ finite_spanning_sets_in.of_le h⟩
+
+section disjointed
+
+variable [measurable_space α]
+
+/-- Given `S : μ.finite_spanning_sets_in {s | measurable_set s}`,
+`finite_spanning_sets_in.disjointed` provides a `finite_spanning_sets_in {s | measurable_set s}`
+such that its underlying sets are pairwise disjoint. -/
+protected def finite_spanning_sets_in.disjointed {μ : measure α}
+  (S : μ.finite_spanning_sets_in {s | measurable_set s}) :
+   μ.finite_spanning_sets_in {s | measurable_set s} :=
+⟨disjointed S.set, measurable_set.disjointed S.set_mem,
+  λ n, lt_of_le_of_lt (measure_mono (disjointed_subset S.set n)) (S.finite _),
+  S.spanning ▸ Union_disjointed⟩
+
+lemma finite_spanning_sets_in.disjoint_disjointed {μ : measure α}
+  (S : μ.finite_spanning_sets_in {s | measurable_set s}) :
+  pairwise (disjoint on S.disjointed.set) :=
+disjoint_disjointed _
+
```

Comment:
You could probably drop this lemma entirely and just use `disjoint_disjointed` below. If you do want to provide API for `disjointed`, the best API is probably just that `S.disjointed.set = disjointed S.set`.